### PR TITLE
[RDY] Naming & Layout Part 1

### PR DESCRIPTION
**Pull request by TheCycoONE**
This pull request updates CorsixTH to consistently use ISO standard naming conventions as part of the effort to conform to the C++ Core Guidelines.

In particular, type prefixes have been removed from all class variables as per: https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#nl5-avoid-encoding-type-information-in-names

ISO Standard style class and function names are used throughout, which conforms to:
https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#nl8-use-a-consistent-naming-style
https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#nl10-avoid-camelcase

Constants have also been changed to lower snake case conforming to:
https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#nl9-use-all_caps-for-macro-names-only

Future pull requests in this series are planned.

### CODE REVIEW

**Comment by TheCycoONE** [comment]
MSVC failure is from doxygen error that is likely a simple oversight. g++ error is more fundamental; it seems to dislike variables with the same names as classes even when the classes are explicitly namespaced (allowed in clang and msvc)

**Comment by TheCycoONE** [comment]
For reasons I'm not clear on, g++ was only concerned with one of the class / variable name collisions (map in map_scanline_iterator). Still plenty of complaints from MSVC though.

**Comment by TheCycoONE** [comment]
Opening this up to bike shedding now.
It's probably the best time to chime in if you don't like any of the new names, or to point out that I missed something with an old style name.

Note:
`L` is still used as the name of the Lua_state variable - this is a lua convention.
`l_` is still used as a prefix for functions to be called from lua. This is temporary until those functions can be moved into another namespace.
Likewise with the `luaT_` prefix.

I did not touch local variable or parameter names in this PR. It's big enough without them.

File: CorsixTH/Src/th_gfx.cpp [hunk 1]
```diff
@@ -32,49 +32,49 @@ SOFTWARE.
...
-        m_pData = pData;
-        m_iLength = iLength;
+        data = pData;
+        remaining_bytes = iLength;
     }
 
-    const uint8_t *m_pData; ///< Pointer to the remaining data.
-    size_t m_iLength;       ///< Remaining number of bytes.
+    const uint8_t *data; ///< Pointer to the remaining data.
+    size_t remaining_bytes;       ///< Remaining number of bytes.
```
Comment (Alberth289346) [inline CorsixTH/Src/th_gfx.cpp]: Documentation comment alignment is messed up, that needs to be addressed at some point.
While I like vertical alignment, if we want to change convention, now is likely a good time.

File: CorsixTH/Src/th_gfx.cpp [hunk 2]
```diff
@@ -308,104 +308,104 @@ inline static void _setmin(int& iLeft, int iRight)
...
         iLoadedElements++;
         iNumElements--;
     }
     return iFirst;
 }
 
-size_t THAnimationManager::makeListElements(size_t iFirstElement, size_t iNumElements,
+size_t animation_manager::make_list_elements(size_t iFirstElement, size_t iNumElements,
                                             size_t &iLoadedListElements,
                                             size_t iListStart, size_t iListCount)
```
Comment (Alberth289346) [inline CorsixTH/Src/th_gfx.cpp]: Indenting of other arguments is incorrect now

File: CorsixTH/Src/th_gfx.cpp [hunk 3]
```diff
@@ -663,179 +663,179 @@ bool THAnimationManager::loadCustomAnimations(const uint8_t* pData, size_t iData
...
-    *pY = m_vFrames[iFrame].iSecondaryMarkerY;
+    *pX = frames[iFrame].secondary_marker_x;
+    *pY = frames[iFrame].secondary_marker_y;
     return true;
 }
 
-bool THAnimationManager::hitTest(size_t iFrame, const THLayers_t& oLayers,
+bool animation_manager::hit_test(size_t iFrame, const ::layers& oLayers,
                                  int iX, int iY, uint32_t iFlags,
                                  int iTestX, int iTestY) const
```
Comment (Alberth289346) [inline CorsixTH/Src/th_gfx.cpp]: Ha, this works, just find a new name that is equally long!  :D

File: CorsixTH/Src/th_gfx.h [hunk 4]
```diff
@@ -484,129 +489,129 @@ class THAnimationManager
...
+    void set_layers_from(const animation_base *pSrc) {layers = pSrc->layers;}
 
    // bool isMultipleFrameAnimation() { return false;}
 protected:
     //! X position on tile (not tile x-index)
-    int m_iX;
+    int x;
     //! Y position on tile (not tile y-index)
-    int m_iY;
+    int y;
```
Comment (Alberth289346) [inline CorsixTH/Src/th_gfx.h]: Perhaps a bit more expressing what the comment says, `xpos_on_tile` or `xpos_within_tile` or so?
Apparently this is currently confusing.

File: CorsixTH/Src/th_gfx.h [hunk 5]
```diff
@@ -484,129 +489,129 @@ class THAnimationManager
...
+    size_t animation_index; ///< Animation number.
+    size_t frame_index;     ///< Frame number.
     union {
         struct {
-            //! Amount to change m_iX per tick
+            //! Amount to change x per tick
             int x;
-            //! Amount to change m_iY per tick
+            //! Amount to change y per tick
             int y;
```
Comment (Alberth289346) [inline CorsixTH/Src/th_gfx.h]: This becomes ambiguous, the comment points to a different `y` than the variable `y` here, but that difference disappears with your renaming. Arguably, `y` here is wrong, I'd make it `dy` at least, or `dy_per_tick` or so

File: CorsixTH/Src/th_gfx.h [hunk 6]
```diff
@@ -484,129 +489,129 @@ class THAnimationManager
...
+    void draw(render_target* pCanvas, int iDestX, int iDestY);
+    bool hit_test(int iDestX, int iDestY, int iTestX, int iTestY);
 
-    void setSheet(THSpriteSheet* pSheet) {m_pSpriteSheet = pSheet;}
-    void setSpeed(int iX, int iY) {m_iSpeedX = iX, m_iSpeedY = iY;}
-    void setLifetime(int iLifetime);
-    void appendSprite(size_t iSprite, int iX, int iY);
-    bool isDead() const {return m_iLifetime == 0;}
+    void set_sheet(sprite_sheet* pSheet) {sheet = pSheet;}
+    void set_speed(int iX, int iY) {speed_x = iX, speed_y = iY;}
```
Comment (Alberth289346) [inline CorsixTH/Src/th_gfx.h]: a comma?! :p

File: CorsixTH/Src/th_gfx.cpp (hunk 1 above)
Comment (TheCycoONE) [reply CorsixTH/Src/th_gfx.cpp]: Formatting is my next PR in this series

File: CorsixTH/Src/th_gfx.cpp [hunk 7]
```diff
@@ -297,7 +297,7 @@ bool THAnimationManager::loadFromTHFile(
     @param [inout] iLeft Left value to check and update.
     @param iRight Second value to check.
  */
-inline static void _setmin(int& iLeft, int iRight)
+static void set_left_to_min(int& iLeft, int iRight)
```
[the remark this answers is not in the document]
Comment (TheCycoONE) [reply CorsixTH/Src/th_gfx.cpp]: We need to start a list

File: CorsixTH/Src/th_gfx.cpp (hunk 2 above)
Comment (TheCycoONE) [reply CorsixTH/Src/th_gfx.cpp]: Ok, it'll get clobbered in the format PR by I will fix it now too.

File: CorsixTH/Src/th_gfx.h (hunk 6 above)
Comment (TheCycoONE) [reply CorsixTH/Src/th_gfx.h]: Esoteric C operator for no reason... another on for the list.

File: CorsixTH/Src/th_gfx.h (hunk 4 above)
Comment (Alberth289346) [reply CorsixTH/Src/th_gfx.h]: In freerct, I used 256 sub-tile positions, multiplied the tile index by 256, and merged both numbers so I had a single position number that expressed the precise position in the game.

File: CorsixTH/Src/th.h [hunk 8]
```diff
@@ -26,59 +26,63 @@ SOFTWARE.
...
 
 //! Generic linked list class (for inheriting from)
-struct THLinkList
+class link_list
 {
-    THLinkList() ;
-    ~THLinkList();
+public:
+    link_list() ;
+    ~link_list();
```
Comment (Alberth289346) [inline CorsixTH/Src/th.h]: Given its purpose is inheriting (as stated in the comment), destructor should be virtual

File: CorsixTH/Src/persist_lua.cpp [hunk 9]
```diff
@@ -406,40 +401,40 @@ class LuaPersistBasicWriter : public LuaPersistWriter
...
                 if(iTable != 3)
                     goto table_resume;
                 break; }
 
             case LUA_TFUNCTION:
                 if(lua_iscfunction(L, 2))
                 {
-                    setErrorObject(2);
-                    setError("Cannot persist C functions");
+                    set_error_object(2);
```
Comment (Alberth289346) [inline CorsixTH/Src/persist_lua.cpp]: Is 2 a magic number that should be eliminated?

File: CorsixTH/Src/persist_lua.cpp [hunk 10]
```diff
@@ -450,37 +445,37 @@ class LuaPersistBasicWriter : public LuaPersistWriter
...
                     lua_getfenv(L, 2);
-                    writeStackObject(-1);
+                    write_stack_object(-1);
                     lua_pop(L, 1);
                 }
                 break;
 
             case LUA_TUSERDATA:
-                if(!_checkThatUserdataCanBeDepersisted(2))
+                if(!check_that_userdata_can_be_depersisted(2))
```
Comment (Alberth289346) [inline CorsixTH/Src/persist_lua.cpp]: bit long, but probably fine

File: CorsixTH/Src/persist_lua.cpp [hunk 11]
```diff
@@ -555,7 +548,7 @@ class LuaPersistBasicWriter : public LuaPersistWriter
             {
                 if(lua_objlen(L, iIndex) != 0)
                 {
-                    setError("Can only persist non-empty userdata if "
+                    set_error("Can only persist non-empty userdata if "
                         "they have a __depersist_size metafield");
```
Comment (Alberth289346) [inline CorsixTH/Src/persist_lua.cpp]: Should such multi-line strings be merged onto a single line at some point?

File: CorsixTH/Src/th_map.h [hunk 12]
```diff
@@ -154,32 +154,32 @@ struct th_map_node_flags
...
+enum class temperature_theme {
     red,         //!< Default warmth colouring (red gradients)
-    multiColour, //!< Different colours (blue, green, red)
-    yellowRed    //!< Gradients of yellow, orange, and red
+    multi_colour, //!< Different colours (blue, green, red)
+    yellow_red    //!< Gradients of yellow, orange, and red
 };
 
-struct THMapNode : public link_list
+struct map_node : public link_list
```
Comment (Alberth289346) [inline CorsixTH/Src/th_map.h]: `link_list` is a class, make this one too?

File: CorsixTH/Src/th_map.h [hunk 13]
```diff
@@ -381,60 +381,61 @@ class THMap
...
     //! \param pNode A pointer to the current node being tested.
-    //! \param prevTemp The array index into THMapNode::aiTemperature that currently stores the temperature of the node (prior to this calculation).
+    //! \param prevTemp The array index into map_node::temperature that currently stores the temperature of the node (prior to this calculation).
     //! \return The weight of the connection, 0 if there is no neighbour, 1 through walls, and 4 through air.
-    uint32_t thermalNeighbour(uint32_t &iNeighbourSum, bool canTravel, std::ptrdiff_t relative_idx, THMapNode* pNode, int prevTemp) const;
+    uint32_t thermal_neighbour(uint32_t &iNeighbourSum, bool canTravel, std::ptrdiff_t relative_idx, map_node* pNode, int prevTemp) const;
 
     //! Create the adjacency matrix if it doesn't already exist
-    void _makeAdjacencyMatrix();
+    void mae_adjacency_matrix();
```
Comment (Alberth289346) [inline CorsixTH/Src/th_map.h]: typo "mae"

File: CorsixTH/Src/th_movie.h [hunk 14]
```diff
@@ -321,70 +321,71 @@ class THMovie
...
     //!
     //! \param pFrame An empty frame which gets populated by the data in the
     //! video packet queue.
     //! \returns 1 if the frame was received, 0 if it was not, and < 0 on error
-    int getVideoFrame(AVFrame *pFrame);
+    int get_video_frame(AVFrame *pFrame);
 #endif
 
-    SDL_Renderer *m_pRenderer; ///< The renderer to draw to
+    SDL_Renderer *renderer; ///< The renderer to draw to
```
Comment (Alberth289346) [inline CorsixTH/Src/th_movie.h]: th_gfx_font has code like `render_target* pCanvas`, while here the `*` is near the variable.
I prefer the latter, as it matches the true semantics of the pointer declaration, but at least it's inconsistent.

File: CorsixTH/Src/iso_fs.cpp [hunk 15]
```diff
@@ -370,78 +370,78 @@ IsoFilesystem::file_handle_t IsoFilesystem::findFile(const char* sPath) const
...
     else
-        return m_pFiles[iFile - 1].iSize;
+        return files[iFile - 1].size;
 }
 
-bool IsoFilesystem::getFileData(file_handle_t iFile, uint8_t *pBuffer)
+bool iso_filesystem::get_file_data(file_handle iFile, uint8_t *pBuffer)
 {
-    if(iFile <= 0 || static_cast<size_t>(iFile) > m_iNumFiles)
+    if(iFile <= 0 || static_cast<size_t>(iFile) > file_count)
```
Comment (Alberth289346) [inline CorsixTH/Src/iso_fs.cpp]: As a general policy, would it make sense to have this 0 vs 1 stuff in the Lua <-> C++ bridge only?

File: CorsixTH/Src/th_gfx_font.h [hunk 16]
```diff
@@ -206,80 +206,80 @@ class THFreeTypeFont final : public THFont
...
-        int iNumRows;
+        int row_count;
 
         //! Alignment of the message in the box
-        eTHAlign eAlign;
+        align alignment;
 
-        //! True when the pData reflects the sMessage given the size constraints
-        bool bIsValid;
+        //! True when the data reflects the message given the size constraints
```
Comment (Alberth289346) [inline CorsixTH/Src/th_gfx_font.h]: Such a shame that we loose the link to the variables by the rename operation, doxygen would need a `#` prefix (or a full `classname::` prefix) to preserve the link.

File: CorsixTH/Src/th_gfx_sdl.cpp [hunk 17]
```diff
@@ -1512,49 +1512,49 @@ void line::depersist(lua_persist_reader *pReader)
...
     {
-        SDL_DestroyTexture(pCacheEntry->pTexture);
-        pCacheEntry->pTexture = nullptr;
+        SDL_DestroyTexture(pCacheEntry->texture);
+        pCacheEntry->texture = nullptr;
     }
 }
 
-void THFreeTypeFont::_makeTexture(render_target *pEventualCanvas, cached_text_t* pCacheEntry) const
+
```
Comment (Alberth289346) [inline CorsixTH/Src/th_gfx_sdl.cpp]: additional empty line

File: CorsixTH/Src/th_lua.h [hunk 18]
```diff
@@ -205,19 +205,19 @@ template <> struct luaT_classinfo<raw_bitmap> {
     static inline const char* name() {return "RawBitmap";}
 };
 
-class THFont;
-template <> struct luaT_classinfo<THFont> {
+class font;
+template <> struct luaT_classinfo<font> {
```
Comment (Alberth289346) [inline CorsixTH/Src/th_lua.h]: could merge the class declaration in the template, I think

File: CorsixTH/Src/th_gfx_font.h [hunk 19]
```diff
@@ -28,38 +28,38 @@ SOFTWARE.
 #include FT_FREETYPE_H
 #endif
 
-enum class eTHAlign {
+enum class align {
```
Comment (Alberth289346) [inline CorsixTH/Src/th_gfx_font.h]: `text_align` ?

File: CorsixTH/Src/th_map_overlays.h [hunk 20]
```diff
@@ -64,13 +64,13 @@ class map_typical_overlay : public map_overlay
     virtual ~map_typical_overlay();
 
     void set_sprites(sprite_sheet* pSheet, bool bTakeOwnership);
-    void set_font(THFont* font, bool take_ownership);
+    void set_font(::font* font, bool take_ownership);
```
Comment (Alberth289346) [inline CorsixTH/Src/th_map_overlays.h]: the need for `::` prefix signals to me that `font` may be too generic as class name.

File: CorsixTH/Src/run_length_encoder.cpp [hunk 21]
```diff
@@ -25,79 +25,79 @@ SOFTWARE.
 #include <new>
 #include <algorithm>
 
-IntegerRunLengthEncoder::IntegerRunLengthEncoder()
+integer_run_length_encoder::integer_run_length_encoder()
```
Comment (Alberth289346) [inline CorsixTH/Src/run_length_encoder.cpp]: `integer` has no meaning does it?
Also `run_length` could be considered an implementation detail, but that somewhat depends on what other encoders we have.

File: CorsixTH/Src/run_length_encoder.cpp [hunk 22]
```diff
@@ -25,79 +25,79 @@ SOFTWARE.
...
+bool integer_run_length_encoder::initialise(size_t iRecordSize)
 {
-    _clean();
-    m_iRecordSize = iRecordSize;
+    clean();
+    record_size = iRecordSize;
 
     // Buffer must hold at least 7 + 2 * 8 records, as the maximum object size
     // is 8 records, 2 of which are needed to detect a repeat, and 7 for the
     // offset at which the objects are found.
```
Comment (Alberth289346) [inline CorsixTH/Src/run_length_encoder.cpp]: so this sets requirements for `iRecordSize`. Imhi a better solution is to silently fix any given record size to the requirements.

File: CorsixTH/Src/run_length_encoder.cpp [hunk 23]
```diff
@@ -129,199 +129,199 @@ void IntegerRunLengthEncoder::_flush(bool bAll)
...
+        != buffer[(iObjIdx2 + i) % buffer_capacity])
         {
             return false;
         }
     }
     return true;
 }
 
-bool IntegerRunLengthEncoder::_output(size_t iObjSize, size_t iObjCount)
+bool integer_run_length_encoder::move_object_to_output(size_t iObjSize, size_t iObjCount)
```
Comment (Alberth289346) [inline CorsixTH/Src/run_length_encoder.cpp]: encoder shouldn't store the output imho

File: CorsixTH/Src/th_lua.h [hunk 24]
```diff
@@ -227,8 +227,8 @@ template <> struct luaT_classinfo<layers> {
     static inline const char* name() {return "Layers";}
 };
 
-class THPathfinder;
-template <> struct luaT_classinfo<THPathfinder> {
+class pathfinder;
```
Comment (Alberth289346) [inline CorsixTH/Src/th_lua.h]: `path_finder` is perhaps better in line with the other code??

File: CorsixTH/Src/th_pathfind.h [hunk 25]
```diff
@@ -121,129 +121,129 @@ class BasePathing
...
 
-    inline bool findIdleTile(const level_map* pMap, int iStartX, int iStartY, int iN)
+    inline bool find_idle_tile(const level_map *pMap, int iStartX, int iStartY, int iN)
     {
-        return m_oIdleTileFinder.findIdleTile(pMap, iStartX, iStartY, iN);
+        return idle_tile_finder.find_idle_tile(pMap, iStartX, iStartY, iN);
     }
 
-    inline bool findPathToHospital(const level_map* pMap, int iStartX, int iStartY)
+    inline bool find_path_to_hospital(const level_map *pMap, int iStartX, int iStartY)
```
Comment (Alberth289346) [inline CorsixTH/Src/th_pathfind.h]: Always fun, these `* ` to ` *` conversions :p

File: CorsixTH/Src/th_sound.cpp [hunk 26]
```diff
@@ -26,49 +26,49 @@ SOFTWARE.
...
+    if(iDataLength < sizeof(uint32_t) + sizeof(th_header))
         return false;
 
     uint32_t iHeaderPosition = reinterpret_cast<const uint32_t*>(pData + iDataLength)[-1];
-    if(static_cast<size_t>(iHeaderPosition) >= iDataLength - sizeof(th_header_t))
+    if(static_cast<size_t>(iHeaderPosition) >= iDataLength - sizeof(th_header))
         return false;
 
-    m_oHeader = *reinterpret_cast<const th_header_t*>(pData + iHeaderPosition);
+    header = *reinterpret_cast<const th_header*>(pData + iHeaderPosition);
```
Comment (Alberth289346) [inline CorsixTH/Src/th_sound.cpp]: `th_header` looks wrong, I guess this is the weak point of not using camel case for types. I'd go for renaming the class to something more specific, eg `sound_header` ?

File: CorsixTH/Src/th_sound.cpp [hunk 27]
```diff
@@ -26,49 +26,49 @@ SOFTWARE.
...
 
-    m_pFiles = reinterpret_cast<th_fileinfo_t*>(m_pData + m_oHeader.table_position);
-    m_iFileCount = m_oHeader.table_length / sizeof(th_fileinfo_t);
+    sound_files = reinterpret_cast<th_fileinfo*>(data + header.table_position);
+    sound_file_count = header.table_length / sizeof(th_fileinfo);
     return true;
 }
 
-size_t THSoundArchive::getSoundCount() const
+size_t th_sound_archive::get_sound_count() const
```
Comment (Alberth289346) [inline CorsixTH/Src/th_sound.cpp]: `get_number_of_sounds` ?

File: CorsixTH/Src/th_sound.cpp [hunk 28]
```diff
@@ -150,179 +150,179 @@ size_t THSoundArchive::getSoundDuration(size_t iIndex)
...
-SDL_RWops* THSoundArchive::loadSound(size_t iIndex)
+SDL_RWops* th_sound_archive::load_sound(size_t iIndex)
 {
-    if(iIndex >= m_iFileCount)
+    if(iIndex >= sound_file_count)
         return nullptr;
 
-    th_fileinfo_t *pFile = m_pFiles + iIndex;
-    return SDL_RWFromConstMem(m_pData + pFile->position, pFile->length);
+    th_fileinfo *pFile = sound_files + iIndex;
```
Comment (Alberth289346) [inline CorsixTH/Src/th_sound.cpp]: th prefix looks wrong/inconsistent

File: CorsixTH/Src/th_sound.cpp [hunk 29]
```diff
@@ -150,179 +150,179 @@ size_t THSoundArchive::getSoundDuration(size_t iIndex)
...
 {
-    return ms_pSingleton;
+    return singleton;
 }
 
-void THSoundEffects::setSoundArchive(THSoundArchive *pArchive)
+void sound_player::populate_from(th_sound_archive *pArchive)
 {
-    for(size_t i = 0; i < m_iSoundCount; ++i)
+    for(size_t i = 0; i < sound_count; ++i)
```
Comment (Alberth289346) [inline CorsixTH/Src/th_sound.cpp]: looks hopelessly silly, the archive should know the number of sounds, or it should be provided as parameter

File: CorsixTH/Src/th_sound.cpp [hunk 30]
```diff
@@ -150,179 +150,179 @@ size_t THSoundArchive::getSoundDuration(size_t iIndex)
...
-    ms_pSingleton = this;
-    m_iCameraX = 0;
-    m_iCameraY = 0;
-    m_fCameraRadius = 1.0;
-    m_fMasterVolume = 1.0;
-    m_fSoundEffectsVolume = 0.5;
-    m_iPostionlessVolume = MIX_MAX_VOLUME;
-    m_bSoundEffectsOn = true;
+    sounds = nullptr;
+    sound_count = 0;
```
Comment (Alberth289346) [inline CorsixTH/Src/th_sound.cpp]: why does a sound player need to know how many sounds exist

File: CorsixTH/Src/th_sound.cpp [hunk 31]
```diff
@@ -150,179 +150,179 @@ size_t THSoundArchive::getSoundDuration(size_t iIndex)
...
 
-int THSoundEffects::reserveChannel()
+int sound_player::reserve_channel()
 {
     // NB: Callers ensure that m_iChannelStatus != 0
     int iChannel = 0;
-    for(; (m_iChannelStatus & (1 << iChannel)) == 0; ++iChannel) {}
-    m_iChannelStatus &=~ (1 << iChannel);
+    for(; (available_channels_bitmap & (1 << iChannel)) == 0; ++iChannel) {}
+    available_channels_bitmap &=~ (1 << iChannel);
```
Comment (Alberth289346) [inline CorsixTH/Src/th_sound.cpp]: always nice if there is no channel available

File: CorsixTH/Src/xmi2mid.cpp [hunk 32]
```diff
@@ -31,75 +31,75 @@ SOFTWARE.
 /*!
     Utility class for reading or writing to memory as if it were a file.
 */
-class MemoryBuffer
+class memory_buffer
```
Comment (Alberth289346) [inline CorsixTH/Src/xmi2mid.cpp]: looks like something that should be shared and re-used at every point where we have a buffer

File: CorsixTH/Src/th_lua_ui.cpp [hunk 33]
```diff
@@ -25,9 +25,9 @@ SOFTWARE.
 #include "th_map.h"
 #include <algorithm>
 
-struct THWindowBase_t {};
+class abstract_window {};
```
Comment (Alberth289346) [inline CorsixTH/Src/th_lua_ui.cpp]: not sure about `abstract_window`, sounds more like something that deals with rectangular areas of some size with generic operations on it, rather than a base class that provides common functionality for a gui, but ymmv.

File: CorsixTH/Src/th_lua_strings.cpp [hunk 34]
```diff
@@ -51,7 +51,7 @@ SOFTWARE.
     environment.
 */
 
-struct THStringProxy_t {};
+class string_proxy {};
```
Comment (Alberth289346) [inline CorsixTH/Src/th_lua_strings.cpp]: It doesn't look like a proxy, much more like a string computation storage, which I totally do not get.

File: CorsixTH/Src/th_lua.h [hunk 35]
```diff
@@ -277,8 +277,8 @@ template <> struct luaT_classinfo<string_proxy> {
     static inline const char* name() {return "StringProxy";}
 };
 
-struct THLfsExt;
-template <> struct luaT_classinfo <THLfsExt> {
+class lfs_ext;
```
Comment (Alberth289346) [inline CorsixTH/Src/th_lua.h]: looks very short compared to the other names

**Comment by Alberth289346** [comment]
I went through the changes in varying states of awakiness, hopefully I covered most of the problems. Lots of comments are not even about the names, I think, at some point I just started writing things that looked wrong, so feel free to move them to the list, or even totally ignore them.
The same somewhat applies to renaming suggestions, in the end, as long as the name is unique, it's fine. There are likely more serious things wrong that need fixing than a name.

File: CorsixTH/Src/th_movie.h (hunk 14 above)
Comment (TheCycoONE) [reply CorsixTH/Src/th_movie.h]: I will be switching all of these to C++ style in the format PR.

https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#nl18-use-c-style-declarator-layout

http://www.stroustrup.com/bs_faq2.html#whitespace

File: CorsixTH/Src/th_lua.h (hunk 35 above)
Comment (TheCycoONE) [reply CorsixTH/Src/th_lua.h]: Extensions to luafilesystem. Ok I will change this to be clearer

File: CorsixTH/Src/th_gfx_font.h [hunk 36]
```diff
@@ -28,38 +28,38 @@ SOFTWARE.
...
     left = 0,
     center = 1,
     right = 2,
 };
 
 /** Structure for the bounds of a text string that is rendered to the screen. */
-struct THFontDrawArea
+struct text_draw_dimensions
```
[the remark this answers is not in the document]
Comment (TheCycoONE) [reply CorsixTH/Src/th_gfx_font.h]: I struggled with this name a bit (and for awhile I did have text_area_*. But a text_area is usually a container for text and I think it should be emphasized that this is the actual resulting size of the text as rendered.  Open to other suggestions.

[the remark this answers is not in the document]
Comment (Alberth289346) [reply CorsixTH/Src/th_gfx_font.h]: `text_layout`? At least, it looks like layout information. The fuzzy part is that there is no text in the structure, which makes `row_count` and `width` quite weird.

File: CorsixTH/Src/th_lua.h (hunk 35 above)
Comment (Alberth289346) [reply CorsixTH/Src/th_lua.h]: There doesn't seem to be anything "Lua specific" in lfs (eg special handling of lua files or so), just regular fs access for lua. Drop the "lua" part?

File: CorsixTH/Src/run_length_encoder.cpp (hunk 21 above)
Comment (TheCycoONE) [reply CorsixTH/Src/run_length_encoder.cpp]: Perhaps should be uint32_run_length_encoder/decoder, because it encodes / decodes uint32s. Being a RLE affects the generated result and means the corresponding decoder has to be compatible (i.e. not an implementation detail). It's a fairly low level concept so I think the name makes sense (except 'integer').

File: CorsixTH/Src/th_lua.h (hunk 24 above)
Comment (TheCycoONE) [reply CorsixTH/Src/th_lua.h]: Pathfinder is one word in English.

File: CorsixTH/Src/th_sound.cpp (hunk 26 above)
Comment (TheCycoONE) [reply CorsixTH/Src/th_sound.cpp]: Maybe even `sound_dat_file_header`

File: CorsixTH/Src/th_gfx_font.h [hunk 37]
```diff
@@ -28,38 +28,38 @@ SOFTWARE.
...
 {
     //! Number of rows the rendered text spans
```
[the remark this answers is not in the document]
Comment (TheCycoONE) [reply CorsixTH/Src/th_gfx_font.h]: Ok, I will make a point to address this in the format PR

File: CorsixTH/Src/th_gfx_font.h (hunk 16 above)
Comment (TheCycoONE) [reply CorsixTH/Src/th_gfx_font.h]: Ok, I didn't know it generated the link before.

File: CorsixTH/Src/persist_lua.cpp (hunk 9 above)
Comment (TheCycoONE) [reply CorsixTH/Src/persist_lua.cpp]: 2 is a Lua stack index. There are a lot of magic number lua stack indexes in the code - it makes it very hard to follow.

File: CorsixTH/Src/th_lua.h (hunk 35 above)
Comment (TheCycoONE) [reply CorsixTH/Src/th_lua.h]: luafilesystem is the name of a 3rd party module. This is an extension to make it play nicer with Windows.

File: CorsixTH/Src/th_gfx.h (hunk 5 above)
Comment (TheCycoONE) [reply CorsixTH/Src/th_gfx.h]: I'll take a second look at this whole struct. Seems like it confuses what it is with how it's used.

File: CorsixTH/Src/persist_lua.cpp (hunk 11 above)
Comment (TheCycoONE) [reply CorsixTH/Src/persist_lua.cpp]: Why?

File: CorsixTH/Src/iso_fs.cpp (hunk 15 above)
Comment (TheCycoONE) [reply CorsixTH/Src/iso_fs.cpp]: Yes

File: CorsixTH/Src/th_gfx_font.h (hunk 36 above)
[the remark this answers is not in the document]
Comment (TheCycoONE) [reply CorsixTH/Src/th_gfx_font.h]: I like `text_layout`, changed locally. I'll push my changes when I've addressed everything.

File: CorsixTH/Src/th_sound.cpp (hunk 28 above)
Comment (TheCycoONE) [reply CorsixTH/Src/th_sound.cpp]: renamed completely, file here is misleading at best.

File: CorsixTH/Src/th_sound.cpp (hunk 30 above)
Comment (TheCycoONE) [reply CorsixTH/Src/th_sound.cpp]: Because this sound player prepares an SDL sound for every piece of sound data so that they are ready to play, and because they don't use an stl container for sounds.

File: CorsixTH/Src/th_sound.h [hunk 38]
```diff
@@ -28,35 +28,35 @@ SOFTWARE.
 #endif
 
 //! Utility class for accessing Theme Hospital's SOUND-0.DAT
-class THSoundArchive
+class th_sound_archive
```
[the remark this answers is not in the document]
Comment (TheCycoONE) [reply CorsixTH/Src/th_sound.h]: fixed

File: CorsixTH/Src/th_lua_strings.cpp (hunk 34 above)
Comment (TheCycoONE) [reply CorsixTH/Src/th_lua_strings.cpp]: No idea, I didn't follow this one enough to dispute the existing name.

**Comment by TheCycoONE** [comment]
Still working on your suggestions. One other significant change I have made since then is renaming map_node to map_tile and everything related (because they are tiles, and node is needlessly general).

File: CorsixTH/Src/th_lua_ui.cpp (hunk 33 above)
Comment (TheCycoONE) [reply CorsixTH/Src/th_lua_ui.cpp]: either way it's short for "abstract base class", I like abstract better these days, so I'm keeping it :p

File: CorsixTH/Src/th_lua.h (hunk 35 above)
Comment (TheCycoONE) [reply CorsixTH/Src/th_lua.h]: Keeping lfs, that's luafilesystem's internal name. e.g. `lfs.dir(...)`, `lfs.attributes(...)`

**Comment by TheCycoONE** [comment]
Addressed all of the naming issues mentioned by @Alberth289346 as well as some of the formatting issues (I know I missed some).

Code issues are out of scope for this PR.

**Comment by TheCycoONE** [comment]
Ok, let's get this merged since it basically blocks C++ work.

AnimView won't be updated to this naming scheme as it clashes with wxWidgets
